### PR TITLE
[Merged by Bors] - feat: generalize normed instances on `UniformSpace.Completion`

### PR DESCRIPTION
**Pull request by eric-wieser**
Notably, `NormedSpace.to_uniformContinuousConstSMul` is generalized to `BoundedSMul` on `PseudoMetricSpace`s, relaxing the scalar considerably from normed fields.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/NormedSpace/PiTensorProduct/InjectiveSeminorm.lean [hunk 1]
```diff
@@ -211,8 +211,7 @@ theorem injectiveSeminorm_le_projectiveSeminorm :
     existsi PUnit, inferInstance, inferInstance
     ext x
     simp only [Seminorm.zero_apply, Seminorm.comp_apply, coe_normSeminorm]
-    have heq : toDualContinuousMultilinearMap PUnit x = 0 := by ext _
```
Comment (eric-wieser) [inline Mathlib/Analysis/NormedSpace/PiTensorProduct/InjectiveSeminorm.lean:214]: This `have` started specializing to `PUnit.{0}`, which broke the `rw` below. Inlining into the `rw` to force contextual unification seems to do the trick.

File: Mathlib/Analysis/Normed/Module/Completion.lean [hunk 2]
```diff
@@ -66,46 +62,39 @@ theorem norm_toComplL {𝕜 E : Type*} [NontriviallyNormedField 𝕜] [NormedAdd
...
+      simp only [← coe_mul, norm_coe]
+      exact norm_mul_le x y
+
+instance [SeminormedCommRing A] : NormedCommRing (Completion A) where
+  __ : CommRing (Completion A) := inferInstance
+  __ : NormedRing (Completion A) := inferInstance
+
+instance
+    [NormedField 𝕜] [SeminormedCommRing A] [NormedAlgebra 𝕜 A] [UniformContinuousConstSMul 𝕜 A] :
+    NormedAlgebra 𝕜 (Completion A) where
```
Comment (j-loreaux) [inline Mathlib/Analysis/Normed/Module/Completion.lean:91]: the bare `instance` line bugs me stylistically. Care to reformat?

**Review by j-loreaux — COMMENTED** [review]
bors d+

File: Mathlib/Analysis/Normed/Module/Completion.lean (hunk 2 above)
Comment (eric-wieser) [reply Mathlib/Analysis/Normed/Module/Completion.lean:91]: Would you be hainstance ppy with `instNormedAlgebra`?

Comment (eric-wieser) [reply Mathlib/Analysis/Normed/Module/Completion.lean]: It turns out that this last argument was redundant anyway.

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**